### PR TITLE
linter change code cleanup, name changes, resourceData.Set()

### PR DESCRIPTION
**Pull request by seansain**
first round of collector linter changes

### CODE REVIEW

File: sumologic/sumologic_sources.go [hunk 1]
```diff
@@ -274,13 +274,39 @@ func resourceSumologicSourceRead(d *schema.ResourceData, source Source) {
...
-	d.Set("default_date_formats", source.DefaultDateFormats)
-	d.Set("filters", source.Filters)
+	d.Set("default_date_formats", flattenDateFormats(source.DefaultDateFormats))
+	d.Set("filters", flattenFilters(source.Filters))
 	d.Set("cutoff_timestamp", source.CutoffTimestamp)
 	d.Set("cutoff_relative_time", source.CutoffRelativeTime)
 	d.Set("fields", source.Fields)
 }
 
+func flattenDateFormats(v []DefaultDateFormat) []map[string]interface{} {
```
Comment (sumovishal) [inline sumologic/sumologic_sources.go]: What's the rationale for this change?

Comment (seansain) [reply sumologic/sumologic_sources.go]: The linter was complaining about `R004: ResourceData.Set() incompatible value type` for Filter and DefaultDateFormat
